samples/Basic/Basic.WebSvc/Controllers/QueryController.cs

```diff
@@ -37,7 +38,149 @@ public Task<IHttpActionResult> GetMetadata(string application, string service)
 		[Route("query/{application}/{service}/{collection}")]
 		public Task<IHttpActionResult> Query(string application, string service, string collection)
 		{
-			return base.QueryAsync(application, service, collection);
+			return QueryAsync(application, service, collection);
+		}
+
+
+		/// <summary>
+		/// Deletes appropriate key & corresponding value from the given reliable collection in the queryable service.
+		/// SINGLE DELETE:
+		/// - DELETE /query/BasicApp/ProductSvc/products and in Body provide a Json:
+		/// In Body: [{
+		///"Key": "sku-218",
+		/// "PartitionId": "946fd004-37aa-4ea6-94a0-3013d8956fef"
+		///}]
+		/// Record belonging to the key provided in the JSON Body of HTTP Request is located in the given partitionID and removed from it.
+		/// 
+		/// BATCH DELETE (Provide an Array of keys in JSON format inside body of HTTP Delete request to delete them all (with kindness ;) ). 
+		/// -DELETE /query/BasicApp/ProductSvc/products
+		/// In Body: [{
+		///"Key": "sku-218",
+		/// "PartitionId": "946fd004-37aa-4ea6-94a0-3013d8956fef"
+		///},{
+		///"Key": "sku-217",
+		/// "PartitionId": "946fd004-37aa-4ea6-94a0-3013d8956fef"
+		///}]
+		/// 
+		/// DELETE A Key from All Partitions :
+		/// -DELETE /query/BasicApp/ProductSvc/products 
+		/// In Body : 
+		/// [{
+		///"Key": "sku-218"
+		///}]
+		/// 
+		/// </summary>
+
+		[HttpDelete]
+		[Route("query/{application}/{service}/{collection}")]
+		public Task<IHttpActionResult> Delete(string application, string service, string collection,
+			[FromBody] ValueViewModel[] obj)
+		{
+			return base.DeleteAsync(application, service, collection, obj);
+		}
+
+		/// <summary>
```

Comment:
The docs/comments should include a description of what happens if the entity already exists.